feat(/api/unicode/names): add unicode codepoint to name service

sidvishnoi opened the pull request:
Part of https://github.com/speced/respec/issues/4462

- Going forward, adding all API endpoints under /api namespace.
- Grabs data from https://unicode.org/Public/UNIDATA/UnicodeData.txt

Landing order: this PR is independent of the others in the batch (no shared files) and can land on its own once the `/update` auth question is resolved. The overlapping PRs should land as #499, #500, #501, #497, #529, #511.

sidvishnoi committed:
feat(/api/unicode/names): add unicode codepoint to name service

feat(/api/unicode/update): update data from r12a's data
  https://github.com/r12a/shared/blob/gh-pages/code/all-names.js
  Webhook not yet configured

sidvishnoi committed:
use unicode data

marcoscaceres commented on routes/api/unicode/lib/scraper.ts:
```diff
@@ -0,0 +1,89 @@
+  options = { ...defaultOptions, ...options } as Options;
+  const hasUpdated = await updateInputSource();
+  if (!hasUpdated && !options.forceUpdate) {
+    console.log("Nothing to update");
```
```suggestion
    console.log("Nothing to update from Unicode.");
```

marcoscaceres replied:
or something similar... just for a bit more context.

marcoscaceres reviewed (approved):
lgtm

marcoscaceres committed:
Update routes/api/unicode/lib/scraper.ts

marcoscaceres committed:
fix(api/unicode): use import.meta.dirname instead of removed legacyDirname

marcoscaceres committed:
fix(api/unicode): skip blank/malformed lines in parseLine

parseLine read parts[1] unconditionally, throwing on any line without a ';' (e.g. a blank or trailing line) and aborting the whole scrape. Guard for fewer than two fields and skip the line instead.

marcoscaceres commented:
Design question on `routes/api/unicode/index.ts`: `router.post("/update", updateRoute)` is unauthenticated, whereas the sibling update endpoints are gated — e.g. caniuse uses `caniuse.post("/update", authGithubWebhook(env("CANIUSE_SECRET")), updateRoute)`.

As written, anyone can POST `/api/unicode/update` and trigger a full UnicodeData.txt download + disk rewrite on the shared box, which is a cheap resource-abuse / DoS vector. Unicode isn't a GitHub webhook source so there's no HMAC signature to verify, but it might be worth either gating it behind a shared secret (a `UNICODE_SECRET` like the others) or not exposing it as a public route at all (cron/manual trigger only). What's the intended trigger model here?

marcoscaceres committed:
test(api/unicode): add parseLine regression tests

Export parseLine and cover the blank/no-semicolon cases that previously threw (reading parts[1] of undefined), plus the normal and <control> name mappings.
